internals/cli/cli.go

```diff
@@ -268,7 +265,19 @@ func (e *exitStatus) Error() string {
 	return fmt.Sprintf("internal error: exitStatus{%d} being handled as normal error", e.code)
 }
 
-func Run() error {
+type RunOptions struct {
+	ClientConfig *client.Config
+	Logger       logger.Logger
+}
+
+func Run(options *RunOptions) error {
+	if options == nil {
+		options = &RunOptions{
+			ClientConfig: &client.Config{},
+		}
+		_, options.ClientConfig.Socket = getEnvPaths()
```

Comment:
Could it make sense to move the client config out of the zero ```RunOptions``` detector, so that Client Config and Logger is handled in the same way?
```
if options == nil {
		options = &RunOptions{}
}

if options.ClientConfig == nil {
    _, options.ClientConfig.Socket = getEnvPaths()
}
```
I am probably missing something, but it feels to me that if I pass in only a new Logger (and Client Config is a nil pointer), that should still work. In theory it should, because both ```client.New()``` and later ```newDefaultRequester()``` do create a client config if its nil, but the configuration will not work according to me, because at least a valid socket has to be setup if BaseURL is ""?

Reply:
Thanks, that cleans it up nicely. Done :)